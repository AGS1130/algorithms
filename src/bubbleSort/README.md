# Bubble sort using TypeScript
> In this lesson we cover the bubble sort algorithm, how it gets its name, and how to implement it using TypeScript.

* We will go ahead and create a sorting function for bubble sort.
* Bubble sort is a comparison based sorting algorithm, so it will need a comparison function.
* Bubble sort works by looping over the input array twice.
* In each iteration the goal is to *Bubble* the highest ranking value to the end.
* To do this we simply check if the current value on the left is less than the current value on the right

```ts
function bubbleSort<T>(
  arr: T[],
  cmp: (a: T, b: T) => number
): T[] {
  for (let index = 0; index < arr.length; index++) {
    for (let index = 0; index < arr.length; index++) {
      
    }
  }
}
```