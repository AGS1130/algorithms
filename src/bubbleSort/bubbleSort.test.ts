import { bubbleSort } from './bubbleSort';

test('sort already sorted array', () => {
  expect([1, 2, 3, 4]).toMatchSnapshot();
});

test('sort array', () => {
  expect([3, 2, 1, 4]).toMatchSnapshot();
});
