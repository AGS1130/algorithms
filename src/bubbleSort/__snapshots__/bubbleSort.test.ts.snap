// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`sort already sorted array 1`] = `
Array [
  1,
  2,
  3,
  4,
]
`;

exports[`sort already sorted array correctly 1`] = `
Array [
  1,
  2,
  3,
  4,
]
`;

exports[`sort array correctly 1`] = `
Array [
  1,
  2,
  3,
  4,
]
`;

exports[`sort array correctly 2`] = `
Array [
  1,
  2,
  3,
  4,
]
`;
